test/functional/page_objects/settings_page.ts

```diff
@@ -486,6 +486,46 @@ export function SettingsPageProvider({ getService, getPageObjects }: FtrProvider
       await this.clickSaveScriptedField();
     }
 
+    async addRuntimeField(name: string, type: string, script: string) {
+      await this.clickAddField();
+      await this.setFieldName(name);
+      await this.setFieldType(type);
+      if (script) {
+        await this.setFieldScript(script);
+      }
+      await this.clickSaveField();
+      await new Promise((resolve) => setTimeout(resolve, 1000 * 30));
+    }
+
+    async clickAddField() {
+      log.debug('click Add Field');
+      await testSubjects.click('addField');
+    }
+
+    async clickSaveField() {
+      log.debug('click Save');
+      await testSubjects.click('fieldSaveButton');
+    }
+
+    async setFieldName(name: string) {
+      log.debug('set field name = ' + name);
+      await testSubjects.setValue('nameField', name);
+    }
+
+    async setFieldType(type: string) {
+      log.debug('set type = ' + type);
+      await testSubjects.setValue('typeField', type);
+    }
+
+    async setFieldScript(script: string) {
+      log.debug('set script = ' + script);
+      await (await testSubjects.findAll('formRowToggle'))[1].click();
+      await new Promise((resolve) => setTimeout(resolve, 1000 * 5));
+      await browser.pressKeys(browser.keys.TAB);
```

Comment:
Sorry for all the questions. You might have copied patterns from elsewhere that I am not aware of.

The above is the best way to focus the "script" code editor or are you also doing accessibility testing by using the TAB?

Reply:
I was unable to find a way to use a `data-test-subj` to set focus with the monaco text editor. Further, its dom is quite complex and it wasn't clear to me if a given dom element would be a reliable way to interact with the editor moving forward.

Reply:
@dmlemeshko Have you worked with any tests using the monaco text editor? Is there anything within our services/framework that could help with this question?

Reply:
I was going to add another functional test to verify entering an invalid script prevents form submission (related to https://github.com/mattkime/kibana/pull/12). In doing so, I realized this code is not currently working as expected. The toggle is not getting enabled (see comment above). However, once that is addressed, then the following logic does not seem to be correctly adding any values to the editor.